File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    stack = []
    for move in moves:
        for i in range(len(board)):
            cur = board[i][move-1]
            if cur != 0:
                board[i][move-1] = 0
                if stack and stack[-1] == cur:
                    answer += 2
                    stack.pop()
                else:
                    stack.append(cur)
                break
    
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack();
        
        for(int j = 0; j<moves.length; j++){
            int line = moves[j]-1;
            for(int i = 0; i <board.length; i++){
                if(board[i][line] != 0) {
                    int cur = board[i][line];
                    board[i][line] = 0;
                    if (!stack.isEmpty() && stack.peek() == cur) {
                        stack.pop();
                        answer += 2;
                        break;
                    }
                    stack.push(cur);
                    break;
                }
            }
        }
        
        return answer;
    }
}